libraries/ms-bootloader/src/jump.c

```diff
@@ -0,0 +1,12 @@
+#include "jump.h"
+
+noreturn void prv_perform_jump(uint32_t sp, uint32_t pc) {
```

Comment:
You'll need to add back the `__attribute__((naked))` here - it tells the compiler not to generate the standard entry/exit code for functions and just to use the raw assembly. It also will stop CI from complaining about `noreturn`.

Also, this is STM32-specific - you'll need to move it to an `stm32f0xx` folder and add a (no-op) `x86` implementation.